python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -2788,6 +2776,128 @@ def transfer_from(
                 " those files are controlled by the other datastore."
             )
 
+        if not refs:
+            return set(), set()
+
+        # Potentially can be transferring from a chain.
+        datastores = getattr(source_datastore, "datastores", [source_datastore])
```

Comment:
Will this work if a chain includes other chains or do you need to expand it recursively?

Reply:
No. I don't think anyone has every tried to do a chain of a chain...